File: Solutions/Python/Equal_Sum.py
class Solution:

	def equilibrium(self,arr, n): 
    	# code here
    	s = 0
    	leftsum = 0
        for i in range(n):
            s += arr[i]
            
        for j in range(n):
            if s-leftsum-arr[j] == leftsum:
                return "YES"
            leftsum += arr[j]
        
        return "NO"

File: Solutions/Java/Equal_Sum.java
class Solution {
    String equilibrium(int arr[], int n) {
        // code here
        int sum = 0;
        for (int i=0; i<n; i++) {
            sum+=arr[i];
        }
        int leftsum = 0;
        for (int i=0; i<n; i++) {
            if (sum-leftsum-arr[i] == leftsum) {
                return "YES";
            }
            leftsum += arr[i];
        }
        return "NO";
    }
}
